examples/js/archive.js

```diff
@@ -134,12 +134,27 @@ const renderThumbnails = (thumbnails = [], url, fullResImgs) => {
     imageContainer.setAttribute('class', 'row');
   });
 };
+ 
+const loadJSONfromResponse = async (response) => {
+  //filter for JSON files from mapknitter
+  const jsonFiles = response.data.files.filter(e => e.format === 'JSON' && e.name.startsWith('mapknitter'))
+ 
+  if (jsonFiles.length > 0) {
+    const answer = confirm('Saved map state detected! Do you want to load it?')
+    // construct JSON url for the first mapknitter JSON file if user confirms
+    if (answer) {
+    const jsonUrl = `https://archive.org/download/${response.data.metadata.identifier}/${jsonFiles[0].name}`
+    reconstructMapFromJson(jsonUrl)
+    }
+  }
+};
 
 function showImages(getUrl) {
   const url = getUrl.replace('details', 'metadata');
 
   axios.get(url)
-      .then((response) => {
+    .then((response) => {
+        loadJSONfromResponse(response)
```

Comment:
This looks great, but I'm seeing this error when entering the url https://archive.org/details/sanborn-1885

`DistortableCollection.js:226 err TypeError: Cannot read properties of undefined (reading 'length')`

The JSON in this is `mapknitter.json` and the contents are:

```json
[{"id":1504,"src":"https://archive.org/download/sanborn-1885/sanborn-1885.jpg",
"width":7573,
"height":6396,
"image_file_name":"sanborn-1885.jpg",
"nodes":[
{"lat":39.32869582897733,"lon":-120.187548995018},
{"lat":39.32982863334289,"lon":-120.18338084220888},
{"lat":39.3270941035765,"lon":-120.182141661644},
{"lat":39.32596125491416,"lon":-120.18630981445314}
],"cm_per_pixel":10.875006010132818}]
```

What do you think is happening? Ah, look:

`Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'avg_cm_per_pixel')
    at reconstructMapFromJson (archive.js:254:53)`

Maybe we have to check for that first, in case it doesn't exist?

Great work so far!!!

Reply:
Oh alright, I'll add that check 

Although I have an idea, I'd explore it and get back to you